bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/scheduler/ScheduledCompletableFuture.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.core.scheduler;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ScheduledFuture;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * Interface returned by all scheduled jobs. It can be used to wait for the value,
+ * cancel the job or check how much time till the scheduled job will run.
+ *
+ * @author Hilbrand Bouwkamp - initial contribution
+ */
+@NonNullByDefault
+public interface ScheduledCompletableFuture<T> extends ScheduledFuture<T> {
+    /**
+     * @return Returns the {@link CompletableFuture} associated with the scheduled job.
+     */
+    CompletableFuture<T> getPromise();
+}
```

Comment:
new line

Reply:
keep your comments and create a new PR once this one is merged :-)
(note: we cannot do any changes to this PR anymore)

Reply:
Just seen that while skimming through the code. You are somewhat familiar with the code, are you? There is no REST interface for querying the scheduler for currently scheduled futures, is there? If not would you think that a PR for adding REST and console support might make sense? If you think so, I'd elaborate on that further an a separate Issue.

Reply:
No, there is no REST or console support atm.
https://github.com/eclipse/smarthome/issues/581 has been entered in 2014, so yes, adding some management for scheduled jobs is definitely desired (at least for the ones that are scheduled through rules by the user, I would leave "internal" jobs out of the game).